### PR TITLE
fix: use `_comp_compgen` for multiline `COMPREPLY=($(compgen ...))`

### PR DESCRIPTION
**Pull request by akinomyoga**
This is another round for `_comp_compgen`.

Also, word lists are re-folded on column 80 to fit the lines within 79 characters.

### CODE REVIEW

File: completions/lintian [hunk 1]
```diff
@@ -124,8 +124,8 @@ _comp_cmd_lintian()
 
     case "$cur" in
         --*)
-            COMPREPLY=($(compgen -W "$lint_actions $general_opts
-                $behaviour_opts $configuration_opts" -- "$cur"))
+            _comp_compgen -- -W "$lint_actions $general_opts $behaviour_opts
+                $configuration_opts"
```
Comment (scop) [inline completions/lintian:128]: Maybe, per https://github.com/scop/bash-completion/pull/961#discussion_r1186844166
```suggestion
            _comp_compgen -- -W '$lint_actions $general_opts $behaviour_opts
                $configuration_opts'
```

File: completions/mdadm [hunk 2]
```diff
@@ -106,40 +105,37 @@ _comp_cmd_mdadm()
 
     if [[ $cur == -* ]]; then
         if ((cword == 1)); then
-            COMPREPLY=($(compgen -W "$options --assemble --build --create
-                --monitor --grow" -- "$cur"))
+            _comp_compgen -- -W "$options --assemble --build --create --monitor
+                --grow"
```
Comment (scop) [inline completions/mdadm:109]: Maybe, per https://github.com/scop/bash-completion/pull/961#discussion_r1186844166
```suggestion
            _comp_compgen -- -W '$options --assemble --build --create --monitor
                --grow'
```

...and similarly for all modifications in this file below?

File: completions/p4 [hunk 3]
```diff
@@ -19,9 +19,8 @@ _comp_cmd_p4()
     elif ((cword == 2)); then
         case $prev in
             help)
-                COMPREPLY=($(compgen -W "simple commands environment
-                    filetypes jobview revisions usage views $p4commands" \
-                    -- "$cur"))
+                _comp_compgen -- -W "simple commands environment filetypes
+                    jobview revisions usage views $p4commands"
```
Comment (scop) [inline completions/p4:23]: Maybe, per https://github.com/scop/bash-completion/pull/961#discussion_r1186844166
```suggestion
                _comp_compgen -- -W 'simple commands environment filetypes
                    jobview revisions usage views $p4commands'
```

File: completions/tshark [hunk 4]
```diff
@@ -26,8 +26,8 @@ _comp_cmd_tshark()
                     _comp_cmd_tshark__prefs="$("$1" -G defaultprefs 2>/dev/null | command sed -ne 's/^#\{0,1\}\([a-z0-9_.-]\{1,\}:\).*/\1/p' |
                         tr '\n' ' ')"
                 : ${prefix:=}
-                COMPREPLY=($(compgen -P "$prefix" -W "$_comp_cmd_tshark__prefs" \
-                    -- "${cur:${#prefix}}"))
+                _comp_compgen -c "${cur:${#prefix}}" -- -P "$prefix" \
+                    -W "$_comp_cmd_tshark__prefs"
```
Comment (scop) [inline completions/tshark:30]: Maybe, per https://github.com/scop/bash-completion/pull/961#discussion_r1186844166
```suggestion
                    -W '$_comp_cmd_tshark__prefs'
```

File: completions/valgrind [hunk 5]
```diff
@@ -84,8 +84,7 @@ _comp_cmd_valgrind()
                     return
                     ;;
                 \<+([0-9])..+([0-9])\>)
-                    COMPREPLY=($(compgen -W "{${value:1:${#value}-2}}" \
-                        -- "$cur"))
+                    _comp_compgen -- -W "{${value:1:${#value}-2}}"
```
Comment (scop) [inline completions/valgrind:87]: Maybe, per https://github.com/scop/bash-completion/pull/961#discussion_r1186844166
```suggestion
                    _comp_compgen -- -W '{${value:1:${#value}-2}}'
```

**Review by scop — APPROVED** [review]
Nice, pre-approved with some single vs double quote comments.

**Comment by scop** [comment]
(Maybe refactor or feat rather than fix for the commit messages though, your call.)

File: completions/lintian (hunk 1 above)
Comment (akinomyoga) [reply completions/lintian:128]: Ah, OK. These are also essentially the same as https://github.com/scop/bash-completion/pull/972#discussion_r1188402795. I'd later process them.

File: completions/valgrind (hunk 5 above)
Comment (akinomyoga) [reply completions/valgrind:87]: I believe this intentionally makes the brace expansions actually take place.

(This is the second trial of the reply. I did reply to this comment, but my previous reply has disappeared. An incident that the notifications to PRs are not delivered is reported today ([GitHub Status - Incident with Pull Requests](https://www.githubstatus.com/incidents/wzv2wd9zvhpr)). It is not mentioned there, but I also observe that not only PRs but the notifications for Issues are not delivered in my project. It reminds me of the disappeared commits in https://github.com/scop/bash-completion/pull/955#discussion_r1186921818. Maybe the GitHub transactions are unstable this week.)

**Comment by akinomyoga** [comment]
> (Maybe refactor or feat rather than fix for the commit messages though, your call.)

Actually, the rewriting using `_comp_split`/`_comp_compgen` (#948) has been originally planned for the protection against pathname expansions (failglob/nullglob) and word splitting (IFS), so they are `fix` in the sense that they solve the actual problems.

**Pull request merged**